\subsection{PHAROS}
\label{subsec:pharos}

The \href{https://druggablegenome.net/}{Illuminating the Druggable Genome} (IDG) was initially motivated to shed light to 1700 targets from four privileged drug target families \cite{santos2016}: G-Protein Coupled Receptors (GPCRs), kinases, ion channels, and nuclear receptors \cite{pharos2016}. Nevertheless, IDG is moving beyond those 4 families and now considers all 20k human coding genes \cite{pharos2016} based on phylogenecity, function, target development level, disease association, tissue expression, chemical ligand and substrate characteristics, and target-family specific characteristics. They developed the \href{http://drugtargetontology.org}{Drug Target Ontology (DTO)} \cite{lin2017}, also available on \href{http://github.com/DrugTargetOntology/DTO}{GitHub} and the \href{http://bioportal.bioontology.org/ontologies/DTO}{NCBO Bioportal}.

The \href{http://juniper.health.unm.edu/tcrd/}{Target Central Resource Database (TCRD)} is the central resource behind the Illuminating the Druggable Genome Knowledge Management Center (IDG-KMC) and can be downladed  \href{http://juniper.health.unm.edu/tcrd/download/}{here}. A list of the datasources used can be found \href{http://targetcentral.ws/Pharos}{here}.

Among PHAROS resources is \href{http://amp.pharm.mssm.edu/Harmonizome/about}{Harmonizome}, that aims to integrate a wide collection of public, disjoint datasets from multiple, internationally recognised datasets (n=114 in October 2018) from 66 different databases that gather information about genomics, epigenetics, transcriptomics, metabolomics, cell lines, diseases, physical interactions, drugs, and curated biomedical literature about mammalian cells \cite{harmonizome2016}. Actualised statistics of Harmonizome can be found \href{http://amp.pharm.mssm.edu/Harmonizome/about}{elsewhere}. One of their main limitations is that all PHAROS data is preprocessed into the space ${1,-1}$. This means that the p-value values have been binarised according to an arbitrary threshold.